packages/opencensus-core/src/metrics/gauges/gauge.ts

```diff
@@ -0,0 +1,169 @@
+    this.labelKeysLength = labelKeys.length;
+    this.defaultLabelValues = initializeDefaultLabels(this.labelKeysLength);
+  }
+
+  /**
+   * Creates a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this gauge, else returns an
+   * existing Point.
+   *
+   * It is recommended to keep a reference to the Point instead of always
+   * calling this method for manual operations.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {types.Point} The value of single gauge.
+   */
+  getOrCreateTimeSeries(labelValues: LabelValue[]): types.Point {
+    validateArrayElementsNotNull(
+        validateNotNull(labelValues, 'labelValues'), 'labelValue');
+    return this.registerTimeSeries(labelValues);
+  }
+
+  /**
+   * Returns a Point for a gauge with all labels not set, or default
+   * labels.
+   *
+   * @returns {types.Point} The value of single gauge.
+   */
+  getDefaultTimeSeries(): types.Point {
+    return this.registerTimeSeries(this.defaultLabelValues);
+  }
+
+  /**
+   * Removes the TimeSeries from the gauge metric, if it is present. i.e.
+   * references to previous Point objects are invalid (not part of the
+   * metric).
+   *
+   * @param {LabelValue[]} labelValues The list of label values.
+   */
+  removeTimeSeries(labelValues: LabelValue[]): void {
+    validateNotNull(labelValues, 'labelValues');
+    this.registeredPoints.delete(hashLabelValues(labelValues));
+  }
+
+  /**
+   * Removes all TimeSeries from the gauge metric. i.e. references to all
+   * previous Point objects are invalid (not part of the metric).
+   */
+  clear(): void {
+    this.registeredPoints.clear();
+  }
+
+  /**
+   * Registers a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this gauge, else returns an
+   * existing Point.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {types.Point} The value of single gauge.
+   */
+  private registerTimeSeries(labelValues: LabelValue[]): types.Point {
+    const hash = hashLabelValues(labelValues);
+    // return if the specified labelValues is already associated with the point.
+    if (this.registeredPoints.has(hash)) {
+      return this.registeredPoints.get(hash);
+    }
+    if (this.labelKeysLength !== labelValues.length) {
+      throw new Error('Label Keys and Label Values don\'t have same size');
+    }
+
+    const point = new PointEntry(labelValues);
+    this.registeredPoints.set(hash, point);
+    return point;
+  }
+
+  /**
+   * Provides a Metric with one or more TimeSeries.
+   *
+   * @returns {Metric} The Metric.
+   */
+  getMetric(): Metric {
+    if (this.registeredPoints.size === 0) {
+      return null;
+    }
+    const hrtimeLocal = process.hrtime();
+    const timeSeriesList: TimeSeries[] = new Array();
+    this.registeredPoints.forEach((point: types.Point) => {
```

Comment:
Instead of `new Array()` and then `.forEach` you can just use `.map`

Reply:
Done